src/Core/Indexes/QuadTree.cs

```diff
@@ -0,0 +1,400 @@
+﻿namespace AEGIS.Indexes
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    public class QuadTree : ISpatialIndex
+    {
+        protected class QuadTreeNode
+        {
+            /// <summary>
+            /// The region of the node.
+            /// </summary>
+            private Envelope bound;
```

Comment:
The "bound" name and the "region" description are misleading, suggest using "envelope" instead.